SharpCompress/Archive/IArchiveEntry.Extensions.cs

```diff
@@ -90,7 +90,7 @@ public static void WriteTo(this IArchiveEntry archiveEntry, Stream streamToWrite
                 FileInfo nf = new FileInfo(destinationFileName);
                 if (nf.Exists)
                 {
-                    if (options.HasFlag(ExtractOptions.PreserveAttributes))
+                    if (options.HasFlag(ExtractOptions.PreserveFileTime))

```

Comment:
I am pretty sure this change is correct, but it does not fix the bug unfortunately. The `ExtractOptions` enum is marked with the `Flags` attribute, however the values are just sequential integers, but not powers of two, so there is a collision between the values of the flags. Can you check that, assuming you want to really fix that bug here?